datafusion/physical-expr/src/window/aggregate.rs

```diff
@@ -18,37 +18,34 @@
 //! Physical exec for aggregate window function expressions.
 
 use std::any::Any;
-use std::iter::IntoIterator;
 use std::ops::Range;
 use std::sync::Arc;
 
 use arrow::array::Array;
-use arrow::compute::SortOptions;
 use arrow::record_batch::RecordBatch;
 use arrow::{array::ArrayRef, datatypes::Field};
 
-use datafusion_common::Result;
 use datafusion_common::ScalarValue;
-use datafusion_expr::{WindowFrame, WindowFrameUnits};
+use datafusion_common::{DataFusionError, Result};
+use datafusion_expr::{Accumulator, WindowFrame, WindowFrameUnits};
 
-use crate::window::window_expr::reverse_order_bys;
-use crate::window::SlidingAggregateWindowExpr;
-use crate::{expressions::PhysicalSortExpr, PhysicalExpr};
-use crate::{window::WindowExpr, AggregateExpr};
-
-use super::window_frame_state::WindowFrameContext;
+use crate::window::window_expr::{reverse_order_bys, AggregateWindowExpr};
+use crate::window::{
+    PartitionBatches, PartitionWindowAggStates, SlidingAggregateWindowExpr, WindowExpr,
+};
+use crate::{expressions::PhysicalSortExpr, AggregateExpr, PhysicalExpr};
 
 /// A window expr that takes the form of an aggregate function
 #[derive(Debug)]
-pub struct AggregateWindowExpr {
+pub struct PlainAggregateWindowExpr {
```

Comment:
It might help here to explain what is the alternative to a `PlainAggregateWindowExpr ` in the doc comments

Also, since this would be a non trivial API change -- anyone who has a `AggregateWindowExpr` in their code would need to change to `PlainAggregateWindowExpr` I wonder how important do you view the new name? 

Maybe we could leave the struct called `AggregateWindowExpr`